app/services/generate_new_week_assignments.rb

```diff
@@ -16,13 +16,19 @@ def call
     def generate_last_assignment_list
       @last_assignment_list = []
 
-      Employee.all.each do |employee|
+      Employee.find_each do |employee|
         last_employee_assignment = Assignment.where(employee_id: employee.id).order(:week_id).last
```

Comment:
How about this:

```ruby
last_employee_assignment =
  Assignment.where(employee_id: employee.id).order(:week_id).last ||
  Assignment.new(week: @week, employee: employee, created_at: Time.zone.now)
```

Then you can get rid of the rest of this method except for the `sort_by`.